KakapoExample/KakapoTests/KakapoDBTests.swift

```diff
@@ -78,41 +136,47 @@ class KakapoDBTests: QuickSpec {
             }
             
             it("shoud return the expected object after inserting it", closure: {
-                let _ = try? sut.insert(UserFactory(firstName: "Hector", lastName: "Zarco", age:25, id: 90))
+                sut.insert({ (id) -> UserFactory in
+                    return UserFactory(firstName: "Hector", lastName: "Zarco", age:25, id: id)
+                })
                 
-                let user = sut.find(UserFactory.self, id: 90)
+                let user = sut.find(UserFactory.self, id: 0)
                 expect(user?.firstName).to(match("Hector"))
                 expect(user?.lastName).to(match("Zarco"))
-                expect(user?.id) == 90
+                expect(user?.id) == 0
             })
             
             it("should fail when inserting invalid id", closure: {
-                do {
-                    try sut.insert(UserFactory(firstName: "Joan", lastName: "Romano", age:25, id: 100))
-                    try sut.insert(UserFactory(firstName: "Joan", lastName: "Romano", age:25, id: 100))
-                } catch KakapoDBError.InvalidId {
-                    expect(true).to(beTrue())
-                } catch {
-                    expect(false).to(beTrue())
-                }
+                sut.insert({ (id) -> UserFactory in
```

Comment:
I think the point is a normal insert works as expected and then a bad insertion is done -> should throw.

By the way I am already subsribed to https://github.com/Quick/Nimble/pull/248